File: ggp-base/src_py/sample_gamer.py
'''
@author: Sam
'''

import random

from org.ggp.base.util.statemachine import MachineState
from org.ggp.base.util.statemachine.implementation.prover import ProverStateMachine
from org.ggp.base.player.gamer.statemachine import StateMachineGamer

class SamplePythonGamer(StateMachineGamer):

    def getName(self):
        pass
        
    def stateMachineMetaGame(self, timeout):
        pass

    def stateMachineSelectMove(self, timeout):
        moves = self.getStateMachine().getLegalMoves(self.getCurrentState(), self.getRole())
        selection = random.choice(moves)
        return selection
        
    def stateMachineStop(self):
        pass
        
    def stateMachineAbort(self):
        pass        
        
    def getInitialStateMachine(self):
        return ProverStateMachine()

File: src_clj/sample_gamer.clj
; An implementation of a sample random gamer in Clojure.
; -Sam Schreiber

; NOTE: the implicit 'this symbol is bound to the local class.

(ns gamer_namespace)

(defn SampleClojureGamer []
  (proxy [org.ggp.base.player.gamer.statemachine.StateMachineGamer] []
    (getInitialStateMachine []
      (new org.ggp.base.util.statemachine.implementation.prover.ProverStateMachine))

    (stateMachineSelectMove [timeout]
      (.
        (. this (getStateMachine) [])
        (getRandomMove
          (. this (getCurrentState) [])
          (. this (getRole) [])
        )
      )
    )
    
    (stateMachineMetaGame [timeout] ())
    (stateMachineAbort [] ())    
    (stateMachineStop [] ())
    (getName [] ())    
  )
)